src/nut.c

```diff
@@ -123,6 +168,121 @@ static void nut_submit(nut_ups_t *ups, const char *type,
   plugin_dispatch_values(&vl);
 } /* void nut_submit */
 
+static int nut_connect(nut_ups_t *ups){
+#ifdef WITH_UPSCLIENT_27
```

Comment:
The reason it's so long is because it's actually the same function twice but with differing logic, but between compiler #if directives.

If the compiler detects that the UPSCLIENT version is 2.7 or greater it can compile using the first version of the function which allows for calls to upscli_init (and a few other symbols). This is what enables the certificate verification to happen.

If however the compiler detects that the UPSCLIENT version is 2.6 or less, before the upscli_init (and other) symbols are available, we use a different definition that will connect without verification and give warning if the user tries to enforce VERIFYPEER because it is unsupported with that library version.

The two definitions separately themselves are not actually too long (40 and 60 lines of code respectively, give or take). I thought about refactoring them out into their own functions with differing names and then nut_connect will call whichever one based on the compiler flag, but we would still need to hide the function with the cert verification logic behind the compiler flags too.

This would result in a differing set of function symbols being available in our code, based on which version of the upsclient library is present. I dislike this approach, so I think ultimately the code should remain as it is now. I think this is the simplest, most straight forward way that will result in the least confusion about whats going on.